appcenter-analytics/android/src/main/java/com/microsoft/appcenter/reactnative/analytics/AppCenterReactNativeAnalyticsModule.java

```diff
@@ -60,4 +66,30 @@ public void trackEvent(String eventName, ReadableMap properties, Promise promise
             promise.reject(e);
         }
     }
+
+    @ReactMethod
+    public void trackEventForTransmissionTarget(String targetToken, String eventName, ReadableMap properties, Promise promise) {
+        AnalyticsTransmissionTarget transmissionTarget = transmissionTargets.get(targetToken);
+        if (transmissionTarget == null) {
+            promise.reject(new IllegalArgumentException("Invalid transmission target token"));
+            return;
+        }
+        try {
+            transmissionTarget.trackEvent(eventName, ReactNativeUtils.convertReadableMapToStringMap(properties));
+            promise.resolve("");
+        } catch (JSONException e) {
+            promise.reject(e);
+        }
+    }
+
+    @ReactMethod
+    public void getTransmissionTarget(String targetToken, Promise promise) {
+        AnalyticsTransmissionTarget transmissionTarget = Analytics.getTransmissionTarget(targetToken);
+        if (transmissionTarget == null) {
+            promise.reject(new IllegalArgumentException("Transmission target may not be null or empty."));
```

Comment:
I know we did inconsistent behaviors in the past but our other native and .NET SDKs are never throwing exceptions to avoid crashing a PROD app just because of a misconfigured telemetry SDK.

For this one native SDK logs an error and return null target and its documented that way.
We should do the same for JS. New JS code should align with native SDKs, we can fix older promises later (never reject anything in new code).